src/components/LabeledSlider.tsx

```diff
@@ -1,17 +1,16 @@
-import { InputNumber, Slider } from "antd";
+import { InputNumber, Slider, SliderSingleProps } from "antd";
+import { SliderRangeProps } from "antd/es/slider";
 import React, { ReactElement, ReactEventHandler, ReactNode, useRef } from "react";
 import styled, { css } from "styled-components";
 import { clamp } from "three/src/math/MathUtils";
 
 import { numberToStringDecimal, setMaxDecimalPrecision } from "../colorizer/utils/math_utils";
 import { excludeUndefinedValues } from "../colorizer/utils/react_utils";
 
-type LabeledRangeSliderProps = {
+type BaseLabeledSliderProps = {
```

Comment:
I'm confused why these props don't extend the antd props? and why you use the antd props down below? I'll try to read more carefully, but maybe you have a way to explain it quickly?

Reply:
Looking at this code again, this is a refactor I could make! When I first wrote this component I was only interested in a subset of the Ant Slider's props, and since it was only being used in two places I didn't feel the need to provide too much flexibility. I'm also not directly passing the props to the Slider anyways, since I have to do some checks for the marks and step value size.

Is this question about L38-47 or L214?

L38-47 are still extending `BaseLabeledSliderProps`, but they define different additional parameters based on whether the `type` is set to `range` or `value`.

Below on L214, I'm extending the antd props for `sliderProps` to conditionally save either value or range-related properties! I ended up changing this to match the style of the other suggestion you'd made about making shared props for the inputs, so it should be a bit cleaner (if more verbose).